src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -905,21 +906,33 @@ bool ShenandoahControlThread::request_concurrent_gc(ShenandoahGenerationType gen
     _requested_gc_cause = GCCause::_shenandoah_concurrent_gc;
     _requested_generation = generation;
     notify_control_thread();
+
     MonitorLocker ml(&_regulator_lock, Mutex::_no_safepoint_check_flag);
-    ml.wait();
+    while (_mode == none) {
+      bool timeout = ml.wait(5);
```

Comment:
Should this be a globals value? Are there other such "took too long to respond" loops goverened by a timeout variable hat might be open to overloading use here perhaps?

Reply:
I can remove the timeout, I was just being paranoid. With the `regulator_lock` being used correctly, it shouldn't be necessary to have a timeout.